superdesk/auth_server/clients.py

```diff
@@ -0,0 +1,257 @@
+# This file is part of Superdesk.
+#
+# Copyright 2019 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import sys
+import uuid
+import logging
+import bcrypt
+import superdesk
+from superdesk.utils import gen_password
+from superdesk.auth_server.scopes import allowed_scopes
+
+
+logger = logging.getLogger(__name__)
+
+
+class AuthServerClientsResource(superdesk.Resource):
+
+    schema = {
+        'client_id': {
```

Comment:
it could store it as `_id` to avoid additional id and it would have unique index out of the box

Reply:
yes I've thought about that, but `_id` has a restricted format, and I would to have a free form for client id. It easier for some use cases, like it is the case now with testing (I can specify any id easily in the feature).

Reply:
:ok_hand:

Reply:
but would be good to add unique index for it then

Reply:
but thinking about it again, there is not real benefit of free form id, for the tests I'll just use an id which is valid for Mongo.